tests/erlang_tests/test_binary_to_term.erl

```diff
@@ -1136,23 +1137,34 @@ test_encode_resource(OTPVersion) ->
     AlteredResource4 = binary_to_term(AlteredResourceBin4),
     false = AlteredResource4 =:= Resource,
     ok.
+test_encode_process_ref() ->
+    AliasesAvailable = is_atomvm_or_otp_version_at_least("24"),
+    if
+        AliasesAvailable ->
+            ProcessRef = erlang:alias(),
+            ProcessRef = binary_to_term(term_to_binary(ProcessRef)),
+            ok;
+        true ->
+            ok
+    end.
 
 % Some binaries are re-encoded differently on earlier BEAM. Verify
 % term_to_binary(binary_to_term(Bin)) is idempotent on AtomVM and recent OTPs.
 binary_to_term_idempotent(Binary, OTPVersion) ->
     Term = binary_to_term(Binary),
-    case erlang:system_info(machine) of
-        "ATOM" ->
-            Binary = term_to_binary(Term);
-        "BEAM" ->
-            OTPRelease = erlang:system_info(otp_release),
-            if
-                OTPRelease >= OTPVersion -> Binary = term_to_binary(Term);
-                true -> ok
-            end
+    CanCheck = is_atomvm_or_otp_version_at_least(OTPVersion),
+    if
+        CanCheck -> Binary = term_to_binary(Term);
+        true -> ok
     end,
     Term.
 
+is_atomvm_or_otp_version_at_least(OTPVersion) ->
```

Comment:
`get_otp_version/0` already exists in this module.

Also there is a poor usage (fixed in #2025 )
```
    OTPVersion = get_otp_version(),
    if
        OTPVersion =:= atomvm orelse OTPVersion >= 26 ->
            B == erlang:term_to_binary(A);
        true ->
            true
    end.
```

be sure to rather do:
```
    OTPVersion = get_otp_version(),
    if
        OTPVersion >= 26 ->
            B == erlang:term_to_binary(A);
        true ->
            true
    end.
```
as atoms sort higher than integers.